src/App.tsx

```diff
@@ -1,56 +1,122 @@
+  const [isDebouncing, setIsDebouncing] = useState(false);
+
+  let isEntered = false;
+  let currentPerson = '';
+
+  if (query.length > 0) {
+    isEntered = true;
+  } else {
+    isEntered = false;
+  }
+
+  const foundPerson
+   = peopleFromServer.find(people => people.name === selectedPerson.name);
+
+  if (foundPerson !== undefined) {
+    currentPerson = foundPerson?.name;
+  }
+
+  const visibleMovies: Person[] = useMemo(
+    () => peopleFromServer.filter((people) => {
+      return people.name.toLowerCase().includes(query.toLowerCase().trim());
+    }),
+    [peopleFromServer, query],
+  );
+
+  const debounceOnChange
+  = debounce((e: string) => {
+    setQuery(e);
+    setIsDebouncing(false);
+  }, 500);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setIsDebouncing(true);
+    debounceOnChange(event.target.value);
+  };
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {`${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`}
       </h1>
 
       <div className="dropdown is-active">
-        <div className="dropdown-trigger">
-          <input
-            type="text"
-            placeholder="Enter a part of the name"
-            className="input"
-          />
-        </div>
-
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
+        {selectedWord.length > 0 && selectedPerson.name === selectedWord
+          ? (
+            <div className="dropdown-trigger">
+              <input
+                type="text"
+                placeholder="Enter a part of the name"
+                className="input"
+                value={selectedWord}
+                onChange={() => setSelectedWord('')}
+              />
             </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
+          )
+          : (
+            <div className="dropdown-trigger">
+              <input
+                type="text"
+                placeholder="Enter a part of the name"
+                className="input"
+                onClick={() => setSelectedWord('')}
+                onChange={handleInputChange}
+              />
             </div>
+          )}
 
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
+        { isEntered === true
+        && isDebouncing === false
+        && selectedWord.length === 0 && (
```

Comment:
use logical operators, please 
```suggestion
        {isEntered && !isDebouncing && !selectedWord.length && (
```